app/main.py

```
@@ -1 +1,29 @@
-# write your code here
+from typing import List
+
+from app.errors import VaccineError, NotWearingMaskError
+
+from app.cafe import Cafe
+
+
+def go_to_cafe(friends: List[dict], cafe: Cafe) -> str:
+    masks_to_buy = 0
+    has_unvaccinated_friend = False
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            has_unvaccinated_friend = True
+        except NotWearingMaskError:
+            masks_to_buy += 1
+        except (VaccineError, NotWearingMaskError) as e:
+            if isinstance(e, VaccineError):
+                has_unvaccinated_friend = True
+            if isinstance(e, NotWearingMaskError):
+                masks_to_buy += 1
```

Issue: The exception handling here is incorrect. In Python, only the first matching except block is executed, so the combined except block (line 19) will never be reached if either of the previous excepts (lines 15 or 17) match. This means you are not correctly accumulating both vaccine and mask errors for all friends as required by the task. You should restructure your exception handling to ensure both errors are tracked for each friend.